scripts/tests.sh

```diff
@@ -22,25 +22,39 @@ root_dir=$(pwd)
 
 for test_dir in ${changed_folders[@]}; do
   cd $test_dir
-  if [[ -d "tests/" ]]; then
-    if test -f "Dockerfile"; then
-      docker build -f Dockerfile .
-      local_exit_code=$?
+
+  if test -f "Dockerfile"; then
+    echo building Dockerfile in $test_dir
+    docker build -f Dockerfile . -t test_image
+    container_name=`docker run -d test_image:latest`
+    sleep 2
+    if [ $(docker inspect -f '{{.State.Running}}' $container_name) = "true" ]; then
+      echo container for $test_dir started successfully
+      local_exit_code=0
     else
-      python -m venv .venv
-      pip install pytest pytest-mock
-      pip install -r requirements.txt
-      pytest -s -v tests/
-      local_exit_code=$?
-      deactivate
-    fi
-    if [[ ! $local_exit_code == 0 ]]; then
-      EXIT_CODE=$local_exit_code
-      echo $test_dir failed. local_exit_code = $local_exit_code, exit = $EXIT_CODE
+      echo docker container did not start in $test_dir
+      local_exit_code=1
     fi
+    docker stop $container_name
+    docker image rm test_image:latest --force
+  elif [[ -d "tests/" ]]; then
+    echo running tests in $test_dir
+    python -m venv .venv
+    pip install pytest pytest-mock
```

Comment:
I suggest putting this into `tests/requirements.txt` and having jina installation using `git+https://` in the `tests/requirements.txt`. Please refer to https://jinaai.slack.com/archives/C0183GCN8AK/p1624852670235800

Reply:
Copy-pasting into all those requirements.txt files is a chore. Also keeping them updated, if something changes. I think it's easier to have it centralized to the script, since it does it for every test.